src/api/config/initializers/wrap_parameters.rb

```diff
@@ -19,54 +17,44 @@
   self.include_root_in_json = false
 end
 
-OBSApi::Application.config.middleware.delete 'ActionDispatch::ParamsParser'
+# Redefine JSON parser to use ActiveSupport::HashWithIndifferentAccess
+# Define XML parser, since Rails' default parsers only include JSON. This XML parser also uses ActiveSupport::HashWithIndifferentAccess
+# Context: https://github.com/rails/rails/blob/04972d9b9ef60796dc8f0917817b5392d61fcf09/actionpack/lib/action_dispatch/http/parameters.rb#L35-L46
+original_parsers = ActionDispatch::Request.parameter_parsers
 
-# custom params parser (modified form of ActionDispatch::ParamsParser)
+json_parser = lambda do |raw_post|
+  original_parsers[Mime[:json].symbol].call(raw_post).with_indifferent_access
+end
 
-class MyParamsParser
-  class ParseError < APIError
+xml_parser = lambda do |raw_post|
+  data = Xmlhash.parse(raw_post)
+  if data
+    { xmlhash: data }.with_indifferent_access
+  else
+    {}
   end
+end
+
+new_parsers = original_parsers.merge({ Mime[:json].symbol => json_parser, Mime[:xml].symbol => xml_parser })
 
+ActionDispatch::Request.parameter_parsers = new_parsers
+
+# Custom Rack middleware to prefer XML over HTML whenever the HTTP header HTTP_ACCEPT isn't set in a request.
+# Without this, clients like `osc` attempt to fetch HTML and they end up getting 404s as OBS has been defaulting to XML
+# since the commit 8145e40a6266f40ce771052fd11e372efc85e116 was introduced many, many years ago. Changing all clients
+# is a lot of work, so we stick to this default.
+#
+# Details on HTTP_ACCEPT:
+# https://www.rfc-editor.org/rfc/rfc9110.html#name-accept
+class PreferXmlOverHtml
```

Comment:
I would prefer to move this out of this file into it's own initializer or rename this one....

Reply:
I will move this custom Rack middleware to its own initializer.

Reply:
Done, please have another look.